src/client/Client.cc

```diff
@@ -11942,7 +11943,11 @@ void Client::refresh_snapdir_attrs(Inode *in, Inode *diri) {
   in->gid = diri->gid;
   in->nlink = 1;
   in->mtime = diri->mtime;
-  in->ctime = diri->ctime;
+  SnapRealm *realm = get_snap_realm_maybe(diri->ino);
+  if (realm && !realm->last_modified.is_zero()) {
+    in->mtime = realm->last_modified;
+  }
+  in->ctime = in->mtime;
```

Comment:
I think client side .snap dir's change_attr (`in->change_attr`) shouldn't always track the .snap parent directory's change_attr (`diri->change_attr`), right?  It should ideally track the MDS side .snap dir's change_attr similar to mtime tracking you do here. So maybe we need a change_attr to also be stored in snaprealm on the MDS side?

Reply:
There is no .snap is the mds. If NFS relies only on mtime changes, then incrementing `change_attr` can be dropped.

Reply:
I should've said client side .snap dir's change_attr should track the MDS snap realm change_attr, which we might want to be added in SnapRealm.

AFAIU, NFS kclient relies on change_attr of a directory to invalidate readdir_cache. In NFS-Ganesha, the Ceph FSAL (CephFS's ganesha driver) report directory's stx_version as the directory's change_attr. See, https://github.com/nfs-ganesha/nfs-ganesha/blob/V3.5/src/FSAL/FSAL_CEPH/internal.c#L199